.github/workflows/ci.ml

```diff
@@ -0,0 +1,65 @@
+module List = ListLabels
+
+let skip_test =
+  match Sys.getenv "SKIP_TEST" with
+  | exception Not_found -> false
+  | s -> bool_of_string s
+
+let run cmd args =
+  let cmd = String.concat " " (cmd :: args) in
+  match Sys.command cmd with
+  | 0 -> ()
+  | n ->
+    Printf.eprintf "'%s' failed with code %d" cmd n;
+    exit n
+
+let opam args = run "opam" args
+
+let pin () =
+  let packages =
+    let packages = Sys.readdir "." |> Array.to_list in
+    let packages =
+      List.fold_left packages ~init:[] ~f:(fun acc fname ->
+          if Filename.check_suffix fname ".opam" then
+            let dot = String.rindex fname '.' in
+            String.sub fname 0 dot :: acc
```

Comment:
```suggestion
            Filename.chop_suffix fname ".opam" :: acc
```

Reply:
We need to make the script work on 4.03 unfortunately 😢

Reply:
Wowzers - is chop_suffix really 4.03+! In which case:

```suggestion
            String.sub fname 0 (String.length dot - 5) :: acc
```

Reply:
Hang on a minute, `Filename.chop_suffix` has been there since the dawn of time?

Reply:
You're right. I got it mixed with `Filename.remove_extension`.